source/deployment/cluster.rst

```diff
@@ -134,75 +134,143 @@ A sample configuration for NGINX is provided below.  It assumes you have two Mat
+Manual failover for master database
+```````````````````````````````````
 
-While connection settings are changing there may be a brief moment when writes to the master database will be unsuccessful. The process waits for all existing connections to finish and starts serving new requests with the new connections. End users attempting to send messages while the switch is happening will have an experience similar to losing connection to the Mattermost server.
+If the need arises to switch from the current master database--for example, if it is running out of disk space, or requires maintenance updates, or for other reasons--you can switch Mattermost server to use one of its read replicas as a master database by updating ``DataSource`` in ``config.json``. 
+
+To apply the settings without shutting down the Mattermost server: 
+
+1. Go to **System Console > Configuration** and click **Reload Configuration from Disk** to reload configuration settings for the Mattermost server from ``config.json``. 
+2. Go to **System Console > Database** and click **Recycle Database Connections** to take down existing database connections and set up new connections in the multi-database configuration. 
+
+While the connection settings are changing, there might be a brief moment when writes to the master database are unsuccessful. The process waits for all existing connections to finish and starts serving new requests with the new connections. End users attempting to send messages while the switch is happening will have an experience similar to losing connection to the Mattermost server.
 
 Transparent Failover
-````````````````````````````````````
-The database can be configured for high availability and transparent failover utilizing the existing database technologies.  We recommend MySQL Clustering, Postgres Clustering, or Amazon Aoura.  Database transparent failover is beyond the scope of this documentation.
+````````````````````
+
+The database can be configured for high availability and transparent failover use the existing database technologies. We recommend MySQL Clustering, Postgres Clustering, or Amazon Aoura. Database transparent failover is beyond the scope of this documentation.
 
------
+Update Guide
+------------
+
+Updating configuration without service interruption
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A service interruption is not required for most configuration updates. See `Updating server with service interruption`_ for a list of configuration updates that require a service interruption.
+
+You can apply updates during a period of low load, but if your HA cluster is sized correctly, you can do it at any time. The system downtime is brief, and depends on the number of Mattermost servers in your cluster. Note that you are not restarting the machines, only the Mattermost server applications. A Mattermost server restart generally takes about 5 seconds.
```

Comment:
1) Please check with dev, but I'm not sure this is true: 

>if your HA cluster is sized correctly, you can do it at any time

If you have 5 servers in HA and the system is correctly sized to handle failure of 2 servers, you still need 3 servers to handle throughput--which means you can't run continuous while updating. 

Our update only works when a single server can handle 100% of anticipated load during the update process. 

2) Please double-check with dev, not sure why we say this: 

>The system downtime is brief, and depends on the number of Mattermost servers in your cluster.

My understanding is there should be no downtime.